librz/util/json_parser.c

```diff
@@ -407,3 +413,55 @@ RZ_API const RzJson *rz_json_item(const RzJson *json, size_t idx) {
 	}
 	return NULL;
 }
+
+RZ_API const RzJson *rz_json_get_path(const RzJson *json, const char *path) {
+	const RzJson *js = json;
+	const char *key;
+	int keysize;
+	ut64 index;
+
+	for (;;) {
+		switch (*path++) {
+		case '\0':
+			// js == json means we've not done any access so far
+			return (js == json) ? NULL : js;
+		case '[':
+			// we could check if js->type != RZ_JSON_ARRAY but rz_json_item will
+			// fail in that case anyway
+			key = path;
+			index = (ut64)strtoull(key, (char **)&path, 10);
+			if (key == path || *path != ']') {
+				RZ_JSON_REPORT_ERROR("JSON path: expected ]", path - 1);
+				return NULL;
+			}
+			++path;
+			js = rz_json_item(js, index);
+			if (!js) {
+				return NULL;
+			}
+			break;
+		case '.':
+			key = path;
+			for (keysize = 0; key[keysize]; ++keysize) {
+				if (strchr(".[", key[keysize]))
+					break;
+			}
+			if (keysize == 0) {
+				RZ_JSON_REPORT_ERROR("JSON path: expected key", path - 1);
+				return NULL;
+			}
+			js = rz_json_get_len(js, key, keysize);
+			if (!js) {
+				return NULL;
+			}
+			path = key + keysize;
+			break;
+		default:
+			//TODO report error
+			RZ_JSON_REPORT_ERROR("JSON path: unexpected char", path - 1);
+			return NULL;
+		}
+	}
+	// unreachable code
```

Comment:
Would `rz_warn_if_reached` be the proper thing to use here?

Reply:
Nevermind, I changed the control flow to be more obvious, removing this dead code completely